core/system/src/mpu/vmpu_armv7m.c

```diff
@@ -478,4 +478,28 @@ void vmpu_arch_init_hw(void)
         UVISOR_TACLDEF_DATA | UVISOR_TACL_EXECUTE,
         subregions_disable_mask
     );
+
+    /* On page heap alignments:
+     *
+     * Individual pages in the page heap are protected by subregions.
+     * A page of size 2^N must have its start address aligned to 2^N.
+     * However, for page sizes > 1/8th region size, the start address is
+     * not guaranteed to be aligned to 2^N.
+     *
+     * Example: 2^N = page size, 2^(N-1) = 1/8th SRAM (32kB page size in a 128kB SRAM).
+     *
+     * |           |
+     * +-----------+ <-- uvisor_config.page_end: 0x30000 = 3/8th * 128kB is aligned to 16kB.
+     * | 32kB page |
+     * +-----------+ <-- page start address: 0x30000 - 32kB = 0x10000 is not aligned to 32kB!!
+     * |           |
+     *
+     * Due to these contradicting alignment requirements, it is not possible
```

Comment:
This will be difficult to solve with `1 page = 1 subregions`, due to subregion alignment issues. Larger pages sizes require `1 page = 2^N subregions`. I'll open a GH issue.